src/main.cpp

```diff
@@ -1203,7 +1219,8 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
                 REJECT_INVALID, "bad-tx-overwinter-version-too-low");
         }
         if (tx.nVersionGroupId != OVERWINTER_VERSION_GROUP_ID &&
```

Comment:
I think that these non-contextual nVersionGroupId checks are not all that valuable. I suppose that what this ultimately resolves to is an inclusion check on all the possible valid version group IDs, but we have to check the the value contextually anyway, so what are these early noncontextual checks actually buying us?

Reply:
The thing I was observing is that these noncontextual checks are going to become just a big "or" that includes all of our versions and version group ids. I guess that restricting these to only values that we recognize is more useful in C++ than in Rust, since we now have a version type that excludes values that we don't recognize by construction over on that side.